src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/PrimitiveDataContract.cs

```diff
@@ -4,28 +4,45 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Reflection;
+using System.Runtime.Serialization.Json;
 using System.Xml;
 
 namespace System.Runtime.Serialization
 {
     internal abstract class PrimitiveDataContract : DataContract
     {
-        internal static readonly PrimitiveDataContract NullContract = new NullPrimitiveDataContract();
+        internal static PrimitiveDataContract? s_NullContract;
+        internal static PrimitiveDataContract NullContract
+        {
+            get
+            {
+                if (s_NullContract == null)
```

Comment:
What's the rationale here? Is it to avoid an allocation when NullContract isn't used? Is this any better than:
```c#
internal static PrimitiveDataContract NullContract { get; } = new NullPrimitiveDataContract();
```

Reply:
Not really. NullPrimitiveDataContract ctor was annotated as RequiresUnreferencedCode, and given the fact that s_NullContract was just initialized statically (on a generated static constructor) there is no easy way to annotate that with RequiresUnreferencedCode itself which is why a Property getter was added. That said, in a later iteration we removed the RequiresUnreferencedCode attribute from the ctor of NullPrimitiveDataContract, so I can actually undo this change now and leave it as we had it. I'll take care of that.